src/utp_wrapper.rs

```diff
@@ -47,12 +48,20 @@ impl UtpWrapper {
                                     }
                                 },
                                 Err(TryRecvError::Disconnected) => break 'outer,
-                                Err(TryRecvError::Empty) => break,
+                                Err(TryRecvError::Empty) => {
+                                    attempts += 1;
+                                    if attempts < 5000 {
+                                        break;
+                                    } else {
+                                        break 'outer;
+                                    }
+                                }
                             }
                         }
                     },
                     Err(_) => break,
                 }
+                ::std::thread::sleep(::std::time::Duration::from_millis(1));
```

Comment:
Is this necessary? The `UdpSocket::read_from` blocking call will already put the thread to sleep and avoid unnecessary CPU share. Also, by sleeping the thread like this, you increase the time we're not reading from socket and therefore increase the chance we lose some packet.